test/app-tap/datetime.test.lua

```diff
@@ -1867,6 +1868,51 @@ test:test("totable{}", function(test)
     end
 end)
 
+test:test('totable{} with timezone', function(test)
+    test:plan(6)
+
+    local DEFAULT_TZOFFSET = 0
+    local DEFAULT_TZ = ''
+
+    local MOSCOW_TZOFFSET = 180
+    local MOSCOW_TZ = 'Europe/Moscow'
+
+    local test_cases = {
+        -- Empty timezone values except tz.
+        {
+            dt = {tz = MOSCOW_TZ},
+            expected = {
+                tzoffset = MOSCOW_TZOFFSET,
+                tz = MOSCOW_TZ,
+            }
+        },
+        -- Empty timezone values except tzoffset.
+        {
+            dt = {tzoffset = MOSCOW_TZOFFSET},
+            expected = {
+                tzoffset = MOSCOW_TZOFFSET,
+                tz = '',
+            }
+        },
+        -- Empty timezone values.
+        {
+            dt = {},
+            expected = {
+                tz = DEFAULT_TZ,
+                tzoffset = DEFAULT_TZOFFSET,
+            },
+        },
+    }
+
+    for _, tc in pairs(test_cases) do
+        local dt_table = date.new(tc.dt):totable()
+        test:is(dt_table.tzoffset, tc.expected.tzoffset,
+                ('[tzoffset]: %s == %s'):format(dt_table.tzoffset, tc.expected.tzoffset))
```

Comment:
Nit: line width is more than 80 symbols.

Maybe it is better to add the local variable for the `tc.expected` look-up for better readability.

Also, `dt_table` -> `dtab`  looks pretier to me.
Feel free to ignore.

Reply:
Updated:

```diff
--- a/test/app-tap/datetime.test.lua
+++ b/test/app-tap/datetime.test.lua
@@ -1905,11 +1905,13 @@ test:test('totable{} with timezone', function(test)
     test:plan(#test_cases  * 2)
 
     for _, tc in pairs(test_cases) do
-        local dt_table = date.new(tc.dt):totable()
-        test:is(dt_table.tzoffset, tc.expected.tzoffset,
-                ('[tzoffset]: %s == %s'):format(dt_table.tzoffset, tc.expected.tzoffset))
-        test:is(dt_table.tz, tc.expected.tz,
-                ('[tz]: %s == %s'):format(dt_table.tz, tc.expected.tz))
+        local dtab = date.new(tc.dt):totable()
+        local expected = tc.expected
+        test:is(dtab.tzoffset, expected.tzoffset,
+                ('[tzoffset]: %s == %q'):format(dtab.tzoffset,
+                                                expected.tzoffset))
+        test:is(dtab.tz, expected.tz,
+                ('[tz]: %s == %s'):format(dtab.tz, expected.tz))
     end
 end)
 ```